clients/drcachesim/reader/reader.cpp

```diff
@@ -204,8 +204,12 @@ reader_t::process_input_entry()
             // Look for encoding bits that belong to this instr.
             if (last_encoding_.size > 0) {
                 if (last_encoding_.size != cur_ref_.instr.size) {
-                    ERRMSG("Encoding size %zu != instr size %zu for PC 0x%zx\n",
-                           last_encoding_.size, cur_ref_.instr.size, cur_ref_.instr.addr);
+                    ERRMSG(
+                        "Encoding size %zu != instr size %zu for PC 0x%zx at ord %" PRIu64
+                        " instr %" PRIu64 " last_timestamp=0x%" PRIx64 "\n",
+                        last_encoding_.size, cur_ref_.instr.size, cur_ref_.instr.addr,
+                        get_record_ordinal(), get_instruction_ordinal(),
+                        get_last_timestamp());
                     assert(false);
```

Comment:
I suppose this assert doesn't fail in release build?

Reply:
No, and in fact that caused headaches where we didn't notice a problem until much later.  Maybe this should be made a release-build fatal error too.

Reply:
This PR only adds checks and failure messages. So the underlying issue that may cause the assert to be hit is still there?

Reply:
The record_scheduler_t spitting of encodings from instrs is the underlying issue for what I was using these extra checks to find: I only later found that out.  It was fixed in #6712.  But these checks still seem good to have, hence this PR, both in case of regressions but also this reader failure one must have multiple possible causes b/c #6303 is still open.  Updated the PR description.

Reply:
It actually is not easy to fail here in release build as there is no error propagation path.  We could directly call abort() like the assert does in debug build.  Maybe that is the best approach, better than setting at_eof or sthg artificial and easy to mis-interpret.  I think having an abort in release will help us; it would have saved me hours or possibly days of work.  Done by adding `assert_release_too` which aborts in release too, for this and the encoding error right below.